execute.go

```diff
@@ -17,14 +28,27 @@ func executeHelper(client *Client, method string, body []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	resp, err := client.session.Do(req)
 	if err != nil {
 		return nil, err
 	}
 
-	respbody, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
+	respbody, rerr := io.ReadAll(resp.Body)
+	if rerr != nil {
+		return nil, rerr
+	}
+
+	// If we didnt get 20x response, unmarshal the error body to be able to format
+	// an informative error message. Mayb this should be >= 400, since I doubt redirect
+	// errors are returned.. but to be safe
+	if resp.StatusCode >= 300 {
```

Comment:
And I think we can do `>= 400`. Because there is no error before 400 code in doc.